.github/workflows/snap.yml

```diff
@@ -19,34 +19,55 @@ env:
   DEFAULT_RISK: edge
 
 jobs:
+  # This job is here just to avoid having "greens" in the "snaps" job for 
+  # architectures that aren't really being built and tested
+  prepare-build-matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      arch-list: ${{ steps.matrix.outputs.archs }}
+    steps:
+      - id: matrix
+        run: |
+          if [[ "${{ github.event_name }}" == "pull_request" ]]
+          then
+            archs='{"arch": "amd64"}'
+          else
+            archs='{"arch": "amd64"},{"arch": "arm64"},{"arch": "ppc64el"},{"arch": "armhf"},{"arch": "s390x"}'
+          fi
+          echo "archs={\"include\":[$archs]}" > $GITHUB_OUTPUT
+
   snaps:
     runs-on: ubuntu-latest
+    needs: [prepare-build-matrix]
     strategy:
       fail-fast: false
-      matrix:
-        arch: [amd64, arm64, ppc64el, armhf, s390x]
+      matrix: ${{ fromJSON(needs.prepare-build-matrix.outputs.arch-list) }}
     steps:
       - name: Checkout Pebble repo
         uses: actions/checkout@v4
         with:
           fetch-depth: 0
 
-      - name: Set LP credentials for remote build
+      - name: Set local paths
         id: set-local-paths
+        run: echo "home=$HOME" >> $GITHUB_OUTPUT
+
+      - name: Set LP credentials for remote build
+        if: github.event_name != 'pull_request'
         run: |
           lp_creds_path="$HOME/.local/share/snapcraft/provider/launchpad"
           mkdir -p $lp_creds_path
           echo '${{ secrets.LP_CREDENTIALS }}' > ${lp_creds_path}/credentials
           git config --global user.email "${{ secrets.PEBBLE_DEV_MAILING_LIST}}"
           git config --global user.name "pebble-dev"
 
-          echo "home=$HOME" >> $GITHUB_OUTPUT
-
       - name: Build ${{ env.SNAP_NAME }} Snap for ${{ matrix.arch }}
         id: build-snap
         uses: snapcore/action-build@v1.1.3
+        env:
+          ARGS: ${{ github.event_name != 'pull_request' && format('remote-build --build-for={0} --launchpad-accept-public-upload', matrix.arch) || 'pack -v' }}
         with:
-          snapcraft-args: remote-build --build-for=${{ matrix.arch }} --launchpad-accept-public-upload
+          snapcraft-args: $ARGS
```

Comment:
I don't quite understand why we need this indirection? Why do we need to go via env/ARGS here?

Reply:
Because when we build locally, we don't use `remote-build` and thus the ARGS will differ. With the latest commit this will become less relevant